bin/spark-stats.js

```diff
@@ -50,3 +55,40 @@ console.log('Starting the http server on host %j port %s', HOST, PORT)
 server.listen(PORT, HOST)
 await once(server, 'listening')
 console.log(`http://${HOST}:${PORT}`)
+
+// Set up the Ethereum provider and contract
+const fetchRequest = new ethers.FetchRequest(RPC_URL)
+fetchRequest.setHeader('Authorization', rpcHeaders.Authorization || '')
+const provider = new ethers.JsonRpcProvider(
+  fetchRequest,
+  null,
+  { batchMaxCount: 1 }
+)
+const ieContract = new ethers.Contract(
+  IE_CONTRACT_ADDRESS,
+  JSON.parse(
+    await fs.readFile(
+      fileURLToPath(new URL('../lib/abi.json', import.meta.url)),
+      'utf8'
+    )
+  ),
+  provider
+)
+
+const onTransfer = async (to, amount) => {
+  const transferEvent = { to_address: to, amount }
+  updateDailyFilStats(pgPool, transferEvent)
+}
+
+// Listen for Transfer events from the IE contract
+const it = onContractEvent({
+  contract: ieContract,
+  provider,
+  rpcUrl: RPC_URL,
+  rpcHeaders
+})
+for await (const event of it) {
+  if (event.name === 'Transfer') {
+    await onTransfer(...event.args)
+  }
+}
```

Comment:
Please revert changes in this file. The necessary infrastructure should be already prepared for you in https://github.com/filecoin-station/spark-stats/blob/1c069bac5a87de7b2ea58f395bb338f9abe67947/observer/bin/spark-observer.js. Feel free to make any adjustments in that file as needed.